quetz/dao.py

```diff
@@ -374,7 +374,37 @@ def create_version(
                 info=info,
                 uploader_id=uploader_id,
             )
+
+            all_existing_versions = (
+                self.db.query(PackageVersion)
+                .filter(PackageVersion.channel_name == channel_name)
+                .filter(PackageVersion.package_name == package_name)
+                .order_by(PackageVersion.version_order.asc())
+            )
+
+            new_version = versionorder.VersionOrder(version)
+            for idx, v in enumerate(all_existing_versions):
+                other = versionorder.VersionOrder(v.version)
+                if other <= new_version and v.build_number <= build_number:
+                    version_order = PackageVersion.version_order
+                    (
+                        self.db.query(PackageVersion)
```

Comment:
I think that's what happens here.

The query should look something like:

```
SET TableField = TableField + 1
WHERE blablabla
```

Reply:
so you don't need to repeat the query, you could just update the all_existing_version directly. Like this:

```
all_exiting_version[i].version_order = new_orders[i]
```

this will save extra lookup in the db

Reply:
but I need to have the additional filter condition for version_order >= idx.

Reply:
so why do you have to run this in a loop? wouldn't it be just enough to find the version_order of the new package, , and increment all version numbers equal or greater, and then insert the new version in the db?

Reply:
something like this:

```    
           version_order = 0
            for idx, v in enumerate(all_existing_versions):
                other = versionorder.VersionOrder(v.version)
                if other > new_version:
                    break
                version_order += 0
            
            new_version.version_order = version_order
            db.query(PackageVersion).
                ...
                filter(PackageVersion.version_order > version_order).
                update({PackageVersion.version_order: PackageVersion.version_order + 1})
           db.insert(new_version)
            
    
```